Automattic/vip-go-mu-plugins
# Add alerting to gutenberg-ramp plugin

## Description

Note that this is a **time-sensitive release** ― please do not deploy this Pull-Request on the VIP Go Platform until the **18th** or **19th** of February 2021. Explanation follows.

We are aiming to remove the Gutenberg Ramp plugin, and this Pull-Request is part of this effort. 

This Pull-Request will update the `gutenberg-ramp` plugin. In this update, there is added alerting in the plugin. The aim of this is to make sure VIP is alerted if `gutenberg_ramp_load_gutenberg()` is being called on the VIP Go platform. Usage of function when running via `WP-CLI` will result in alerts, as this is sufficient to detect if any sites continue to invoke the function and should not lead to too many alerts being logged.

On the 17th and 18th of February, we aim to have all Pull-Requests merged that _should_ result in most or all invocations to `gutenberg_ramp_load_gutenberg()` being removed, but we will need to make sure this is really the case. The alerting will help us to determine.

Note that we cannot use GitHub search for this same purpose, as it only searches one branch on each repository, and invocations to `gutenberg_ramp_load_gutenberg()` can be in place on non-indexed branches.

The code-change itself is available here: https://github.com/Automattic/gutenberg-ramp/compare/7d0cd89e5df5efec902cadc56ca0c844d7be4b7a...71c609e07d3bfbd5301f24ab1cd75d70ec993ad8.diff

## Changelog Description

### Plugin Updated: gutenberg-ramp to latest upstream

The `gutenberg-ramp` plugin is deprecated and should not be in regular use. To make this transition easier and to stay aware of all users, we have upgraded `gutenberg-ramp` to the latest upstream version. This upgrade brings alerting such that if sites continue to call the main function of the `gutenberg-ramp` plugin, we will be alerted. The idea being that we want to remove all usage of the plugin and need to know if the plugin is still being used and by whom.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [N/A] This change has relevant unit tests (if applicable).
- [N/A] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

1. Get a diff for this Pull-Request: https://github.com/Automattic/gutenberg-ramp/compare/7d0cd89e5df5efec902cadc56ca0c844d7be4b7a...71c609e07d3bfbd5301f24ab1cd75d70ec993ad8.diff
2. Apply in `gutenberg-ramp` plugin on a VIP Go sandbox.
3. Make sure `gutenberg-ramp` is being used by the site by adding this to `client-mu-plugins/test.php`: `gutenberg_ramp_load_gutenberg(true);`
4. Run `WP-CLI`, observe a message being logged. In case of testing on a regular sandbox, error will be found in `php-errors`.
4. Load the site in a browser, observe no message will be logged.
